server/src/main/java/org/apache/cassandra/sidecar/db/schema/TableHistorySchema.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.cassandra.sidecar.config.ServiceConfiguration;
+import org.apache.cassandra.sidecar.coordination.ExecuteOnClusterLeaseholderOnly;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Schema definition and management for tracking table schema evolution history.
+ * This class extends {@link TableSchema} to provide specialized schema management for storing
+ * historical versions of table schemas in Cassandra Sidecar.
+ */
+@Singleton
+public class TableHistorySchema extends TableSchema implements ExecuteOnClusterLeaseholderOnly
+{
+    private static final String TABLE_SCHEMA_HISTORY = "table_schema_history";
+
+    private final SchemaKeyspaceConfiguration keyspaceConfig;
+
+    // prepared statements
+    private PreparedStatement insertTableSchema;
+    private PreparedStatement selectVersionTableSchema;
+
+    @Inject
+    public TableHistorySchema(ServiceConfiguration configuration)
+    {
+        this.keyspaceConfig = configuration.schemaKeyspaceConfiguration();
+    }
+
+    @Override
+    protected void prepareStatements(@NotNull Session session)
+    {
+        insertTableSchema = prepare(insertTableSchema, session, CqlLiterals.insertTableSchema(keyspaceConfig));
+        selectVersionTableSchema = prepare(selectVersionTableSchema, session, CqlLiterals.selectVersionTableSchema(keyspaceConfig));
+    }
+
+    @Override
+    protected String keyspaceName()
+    {
+        return keyspaceConfig.keyspace();
+    }
+
+    @Override
+    protected String tableName()
+    {
+        return TABLE_SCHEMA_HISTORY;
+    }
+
+    @Override
+    protected boolean exists(@NotNull Metadata metadata)
+    {
+        KeyspaceMetadata ksMetadata = metadata.getKeyspace(keyspaceConfig.keyspace());
+        if (ksMetadata == null)
+        {
+            return false;
+        }
+
+        return ksMetadata.getTable(TABLE_SCHEMA_HISTORY) != null;
+    }
+
+    @Override
+    protected String createSchemaStatement()
+    {
+        return String.format("CREATE TABLE IF NOT EXISTS %s.%s (" +
+                             "  keyspace_name text," +
+                             "  table_name text," +
+                             "  version uuid," +
+                             "  created_at timeuuid," +
+                             "  table_schema text," +
+                             "  PRIMARY KEY ((ks, tb), version)" +
+                             ")",
+                             keyspaceConfig.keyspace(), TABLE_SCHEMA_HISTORY);
+    }
+
+    public PreparedStatement insertTableSchema()
+    {
+        return insertTableSchema;
+    }
+
+    public PreparedStatement selectVersionTableSchema()
+    {
+        return selectVersionTableSchema;
+    }
+
+    private static class CqlLiterals
+    {
+        static String insertTableSchema(SchemaKeyspaceConfiguration config)
+        {
+            return withTable("INSERT INTO %s.%s (ks, tb, version, created_at, table_schema) " +
```

Comment:
after the change above, we should use the new names for the keyspace, table columns
```suggestion
            return withTable("INSERT INTO %s.%s (keyspace_name, table_name, version, created_at, table_schema) " +
```